modules/sample-dropwizard/src/test/scala/core/Dropwizard/RequestBodiesTest.scala

```diff
@@ -0,0 +1,15 @@
+package core.Dropwizard
+
+import componentsRequestBodies.client.dropwizard.definitions.SomeSchema
+import componentsRequestBodies.client.dropwizard.requestBodies.RequestBodiesClient
+import java.math.BigInteger
+import java.net.URI
+import org.scalatest.{FreeSpec, Matchers}
+
+class RequestBodiesTest extends FreeSpec with Matchers {
+  "References to requestBodies should resolve and generate the proper args/methods" in {
```

Comment:
Essentially it's more a "this test itself will not compile if something breaks".

Reply:
Ah. Maybe just don't put it inside `test` then? The `test` context implies runtime behavior. They've got another plugin that lets you check to see if an arbitrary string compiles in your tests, failing with a descriptive error, but I don't remember what it's called. I think just removing it from the test block and adding a comment is sufficient for now.

Reply:
Would it make more sense to do a structure test instead, like you do with scala?  I haven't done that for java yet, but might be worthwhile to set up the ability to do that and make sure it works.

Reply:
imo yes, only if we care about the literal structures changing.

Reasons for preferring this are:

- No need to run the entire compiler to test critical components, so bonus speed
- No need to generate code in order to test the output

Reasons for running the entire compiler:

- Full integration with dependencies and potentially changing external APIs
- Structurally different but functionally equivalent stuff should be judged based on its behaviour, not just the syntax
- Comparing ASTs in scalameta has been a beast due to subtle differences in parser structure (`foo( a => f(a) )`, `foo({ a => f(a) })`, and `foo( a => { f(a) })` all represent different ASTs)

There are reasons for putting code in one place over another; I'll leave it up to you whether you think it belongs here.

Reply:
I guess it's sorta in the middle?  I don't care about the structure being completely exact, but I want to test "The method for the `foo` operation takes a `double` and the returned call builder has a method that takes an optional `BigInteger`" and "The method for the `bar` operation takes a `SomeSchema` argument, and the `SomeSchema` builder constructor takes a string, and the builder has a method that takes an optional `boolean`".

Actually it might even be simpler than that: before I made this change, the generator would error out saying it couldn't resolve the $ref.  So really I just want to make sure that the generator actually generates code, and (for thoroughness) that the generated code compiles.  But because of how the samples runner works, I do need something that actually uses the generated code, because a generator failure doesn't fail the build.

Running the entire compiler _does_ allow me to ensure that structurally different but functionally equivalent stuff is judged on behavior, for the most part, which is why I was doing it here, but writing a "does this just compile" feels wrong as a test.

Agree with the problematic nature of comparing ASTs; I recall how fragile that can be sometimes when having to update some of the scala tests.  I'm not eager to do this here, but it's also likely a much faster test than what I'm doing now.

Still not really sure what to do here.

Reply:
We'll need structure comparison in the future anyway, so if you want the answer to the toss-up, it would be great to have that here.

Reply:
Ok, got structure comparison working.